flink-runtime/src/main/java/org/apache/flink/runtime/deployment/TaskDeploymentDescriptorFactory.java

```diff
@@ -58,30 +60,45 @@
 import java.util.Map;
 import java.util.Optional;
 
+import static org.apache.flink.configuration.ConfigOptions.key;
+import static org.apache.flink.util.Preconditions.checkNotNull;
 import static org.apache.flink.util.Preconditions.checkState;
 
 /**
  * Factory of {@link TaskDeploymentDescriptor} to deploy {@link
  * org.apache.flink.runtime.taskmanager.Task} from {@link Execution}.
  */
 public class TaskDeploymentDescriptorFactory {
+    @Experimental
+    public static final ConfigOption<Integer> OFFLOAD_SHUFFLE_DESCRIPTORS_THRESHOLD =
+            key("jobmanager.deployment.offload-shuffle-descriptors-threshold")
+                    .intType()
+                    .defaultValue(4_000 * 4_000)
+                    .withDescription(
+                            "The threshold to force enable offload shuffle descriptors via blob server."
```

Comment:
Maybe change the first sentence to "Threshold for offloading shuffle descriptors to blob server. Once the number of shuffle descriptors exceeds this value, we will offload the shuffle descriptors to blob server."